### PR TITLE
Improve api doc (2d&UI)

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos/cocos-engine/issues/13888

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/assets/bitmap-font.ts [hunk 1]
```diff
@@ -102,6 +102,9 @@ export class FontAtlas {
  */
 @ccclass('cc.BitmapFont')
 export class BitmapFont extends Font {
+    /**
+     * @deprecated since v3.7
```
Comment (SantyWang) [inline cocos/2d/assets/bitmap-font.ts]: -> Since v3.7.0. and add the reason why we deprecate it.

File: cocos/2d/assets/sprite-atlas.ts [hunk 2]
```diff
@@ -46,13 +46,17 @@ interface ISpriteFrameList {
  */
 @ccclass('cc.SpriteAtlas')
 export class SpriteAtlas extends Asset {
+    /**
+     * @en All sprite frames in the sprite atlas
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-atlas.ts]: miss `.`. please check all comment.

File: cocos/2d/assets/sprite-atlas.ts [hunk 3]
```diff
@@ -46,13 +46,17 @@ interface ISpriteFrameList {
...
+     */
     @serializable
     @editable
     public spriteFrames: ISpriteFrameList = js.createMap();
 
     /**
      * @en Get the [[Texture2D]] asset of the atlas.
-     * @zh 获取精灵图集的贴图。请注意，由于结构调整优化，在 v1.1 版本之前，此函数的返回值为 imageAsset，在 v1.1 版本之后修正为 texture，想要获取 imageAsset 可使用 getTexture().image 获取
+     * @zh 获取精灵图集的贴图。
      * @returns The texture2d asset
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-atlas.ts]: The return value should have i18n too. please check all comments

File: cocos/2d/assets/sprite-frame.ts [hunk 4]
```diff
@@ -500,13 +506,21 @@ export class SpriteFrame extends Asset {
...
+     */
     get packable () {
         return this._packable;
     }
     set packable (value: boolean) {
         this._packable = value;
     }
 
+    /**
+     * @en Original info before packed to dynamic atlas,includes texture and width and height.Before packed to dynamic atlas is null.
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: info -> information. Before packed to dynamic atlas is null. -> it's null before being packed to dynamic atlas.

File: cocos/2d/assets/sprite-frame.ts [hunk 5]
```diff
@@ -770,6 +787,7 @@ export class SpriteFrame extends Asset {
      * @en Resets the sprite frame data
      * @zh 重置 SpriteFrame 数据。
      * @param info SpriteFrame initialization information
+     * @param clearData clear Data before initialization
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: miss `.` and chinese version

File: cocos/2d/assets/sprite-frame.ts [hunk 6]
```diff
@@ -1446,6 +1477,10 @@ export class SpriteFrame extends Asset {
         this._calculateUV();
     }
 
+    /**
+     * @en Check whether the skeleton is validate
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: skeleton -> sprite frame

File: cocos/2d/assets/sprite-frame.ts [hunk 7]
```diff
@@ -240,9 +243,11 @@ const temp_uvs: IUV[] = [{ u: 0, v: 0 }, { u: 0, v: 0 }, { u: 0, v: 0 }, { u: 0,
 @ccclass('cc.SpriteFrame')
 export class SpriteFrame extends Asset {
     /**
-     * @en Create a SpriteFrame object by an image asset or an native image asset
-     * @zh 通过 Image 资源或者平台相关 Image 对象创建一个 SpriteFrame 对象
-     * @param imageSourceOrImageAsset ImageAsset or ImageSource, ImageSource support HTMLCanvasElement HTMLImageElement IMemoryImageSource
+     * @en Create a SpriteFrame object by an image asset or an native image asset.
+     * @zh 通过 Image 资源或者平台相关 Image 对象创建一个 SpriteFrame 对象。
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: SpriteFrame 对象 -> SpriteFrame 资源

File: cocos/2d/assets/sprite-frame.ts [hunk 8]
```diff
@@ -240,9 +243,11 @@ const temp_uvs: IUV[] = [{ u: 0, v: 0 }, { u: 0, v: 0 }, { u: 0, v: 0 }, { u: 0,
...
+     * @param imageSourceOrImageAsset @en ImageAsset or ImageSource, ImageSource support HTMLCanvasElement HTMLImageElement IMemoryImageSource.
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: ImageSource could be HTMLCanvasElement, HTMLImageElement, IMemoryImageSource

File: cocos/2d/assets/sprite-frame.ts [hunk 9]
```diff
@@ -435,7 +440,7 @@ export class SpriteFrame extends Asset {
     }
 
     /**
-     * @en The uuid of the atlas asset, if exist
+     * @en The uuid of the atlas asset, if exist.
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: exist -> exists

File: cocos/2d/assets/sprite-frame.ts [hunk 10]
```diff
@@ -500,64 +508,75 @@ export class SpriteFrame extends Asset {
...
+     */
     get packable () {
         return this._packable;
     }
     set packable (value: boolean) {
         this._packable = value;
     }
 
+    /**
+     * @en Original information before packed to dynamic atlas, includes texture and width and height. It's null before being packed to dynamic atlas.
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: includes texture, width, height

File: cocos/2d/assets/sprite-frame.ts [hunk 11]
```diff
@@ -500,64 +508,75 @@ export class SpriteFrame extends Asset {
...
+     * @zh 精灵自动合图之前的原始 texture 和宽高信息。在参与自动合图之前此值为 null。
+     */
     get original () {
         return this._original;
     }
 
     /**
-     * @en Number of pixels corresponding to unit size in world space (pixels per meter)
-     * @zh 世界空间中的单位大小对应的像素数量（像素每米）
+     * @en Number of pixels corresponding to unit size in world space (pixels per meter).
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: pixels per unit. (像素每单位)

File: cocos/2d/assets/sprite-frame.ts [hunk 12]
```diff
@@ -665,7 +684,9 @@ export class SpriteFrame extends Asset {
      * If it's an atlas texture, a transparent pixel area is proposed for the actual mapping of the current texture.
      * @zh 获取 SpriteFrame 的纹理矩形区域。
      * 如果是一个 atlas 的贴图，则为当前贴图的实际剔除透明像素区域。
-     * @deprecated since v1.2, please use [[rect]]
+     * @param out @en The output rect object @zh 输出的矩形区域。
+     * @returns @en The rect object. @zh 矩形区域对象。
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: The rect object. -> The rect.

File: cocos/2d/assets/sprite-frame.ts [hunk 13]
```diff
@@ -702,17 +726,18 @@ export class SpriteFrame extends Asset {
...
+     * @param size @en The new original size. @zh 新设置的原始大小。
+     * @deprecated since v1.2, please use [[originalSize]].
      */
     public setOriginalSize (size: Size) {
         this.originalSize = size;
     }
 
     /**
-     * @en Returns the offset of the frame
+     * @en Returns the offset of the frame.
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: Returns -> Gets

File: cocos/2d/assets/sprite-frame.ts [hunk 14]
```diff
@@ -725,51 +750,56 @@ export class SpriteFrame extends Asset {
...
+     * @returns @en Gfx Texture resource. @zh GFX 贴图资源。
      */
     public getGFXTexture () {
         return this._texture.getGFXTexture();
     }
 
     /**
-     * @en Gets the sampler resource of its texture
-     * @zh 贴图资源的采样器
+     * @en Gets the sampler resource of its texture.
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: sampler resource -> GFX sampler

File: cocos/2d/assets/sprite-frame.ts [hunk 15]
```diff
@@ -837,9 +867,10 @@ export class SpriteFrame extends Asset {
     }
 
     /**
-     * @en Check whether the rect of the sprite frame is out of the texture boundary
+     * @en Check whether the rect of the sprite frame is out of the texture boundary.
      * @zh 判断精灵计算的矩形区域是否越界。
-     * @param texture
+     * @param texture @en Texture resources for sprite frame. @zh SpriteFrame 的图像资源。
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: 图像资源 -> 贴图资源

File: cocos/2d/assets/sprite-frame.ts [hunk 16]
```diff
@@ -1434,10 +1469,19 @@ export class SpriteFrame extends Asset {
...
     public onLoaded () {
         this._calcTrimmedBorder();
     }
 
+    /**
+     * @en default init.
+     * @zh 默认初始化。
+     * @param uuid @en Asset uuid. @zh 资源 uuid。
+     */
     public initDefault (uuid?: string) {
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts:1487]: deprecated as engine internal interface

File: cocos/2d/assets/sprite-frame.ts [hunk 17]
```diff
@@ -1434,10 +1469,19 @@ export class SpriteFrame extends Asset {
         }
     }
 
+    /**
+     * @en complete loading callback.
+     * @zh 加载完成回调。
+     */
     public onLoaded () {
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts:1477]: deprecated as engine internal interface

File: cocos/2d/assets/sprite-frame.ts [hunk 18]
```diff
@@ -1446,6 +1490,11 @@ export class SpriteFrame extends Asset {
         this._calculateUV();
     }
 
+    /**
+     * @en Check whether the sprite frame is validate.
+     * @zh 检查当前 sprite frame 对象是否是有效的。
+     * @returns @en validate or not. @zh 是否有效。
+     */
     public validate () {
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts:1501]: deprecated as engine internal interface

File: cocos/2d/assets/ttf-font.ts [hunk 19]
```diff
@@ -59,6 +59,11 @@ export class TTFFont extends Font {
         return { uuid: this._uuid, __nativeName__: this._native, ext: path.extname(this._native), __isNative__: true };
     }
 
+    /**
+     * @en default init.
+     * @zh 默认初始化。
+     * @param uuid @en Asset uuid. @zh 资源 uuid。
+     */
     public initDefault (uuid?: string) {
```
Comment (SantyWang) [inline cocos/2d/assets/ttf-font.ts:68]: deprecated as engine internal interface

File: cocos/2d/components/mask.ts [hunk 20]
```diff
@@ -260,6 +260,10 @@ export class Mask extends Component {
         }
     }
 
+    /**
+     * @en Rendering component for providing stencilBuffer information.
+     * @zh 用于提供stencilBuffer 信息的渲染组件。
```
Comment (SantyWang) [inline cocos/2d/components/mask.ts]: 提供stencilBuffer -> 提供 stencil buffer

File: cocos/2d/components/label.ts [hunk 21]
```diff
@@ -189,12 +172,32 @@ ccenum(CacheMode);
...
+     */
     public static HorizontalAlign = HorizontalTextAlignment;
+    /**
+     * @en Enum for vertical text alignment.
+     *
+     * @zh 文本垂直对齐类型。
+     */
     public static VerticalAlign = VerticalTextAlignment;
+    /**
+     * @en Enum for Overflow.
```
Comment (SantyWang) [inline cocos/2d/components/label.ts]: Enum for label overflow mode.

File: cocos/2d/components/mask.ts (hunk 20 above)
Comment (SantyWang) [inline cocos/2d/components/mask.ts]: 提供stencilBuffer -> 提供 stencil buffer

File: cocos/2d/components/mask.ts [hunk 22]
```diff
@@ -319,8 +319,8 @@ export class Mask extends Component {
     }
 
     /**
-     * Hit test with point in World Space.
-     *
+     * @en Hit test with point in World Space.
+     * @zh 世界空间中的点击测试。
      * @param worldPt point in World Space.
```
Comment (SantyWang) [inline cocos/2d/components/mask.ts]: miss chinese version

File: cocos/2d/components/sprite.ts [hunk 23]
```diff
@@ -431,9 +435,25 @@ export class Sprite extends UIRenderer {
...
+     * @zh 填充类型。
+     */
     public static FillType = FillType;
+    /**
+     * @en Enum for sprite type.
+     * @zh Sprite 类型。
+     */
     public static Type = SpriteType;
+    /**
+     * @en Sprite Size can track trimmed size, raw size or none.
```
Comment (SantyWang) [inline cocos/2d/components/sprite.ts]: Sprite's size mode, including trimmed size, raw size, and none.

File: cocos/2d/framework/ui-transform.ts [hunk 24]
```diff
@@ -580,7 +605,7 @@ export class UITransform extends Component {
      * @zh
      * 返回父节坐标系下的轴向对齐的包围盒。
      *
-     * @return - 节点大小的包围盒
+     * @returns @en A "local" axis aligned bounding box of the node.  @zh 本地坐标系下的包围盒。
```
Comment (SantyWang) [inline cocos/2d/framework/ui-transform.ts]: an axis aligned bounding box of this node in local space coordinate.

File: cocos/2d/framework/ui-transform.ts [hunk 25]
```diff
@@ -609,7 +634,7 @@ export class UITransform extends Component {
      * 返回节点在世界坐标系下的对齐轴向的包围盒（AABB）。
      * 该边框包含自身和已激活的子节点的世界边框。
      *
-     * @returns - 返回世界坐标系下包围盒。
+     * @returns @en A "world" axis aligned bounding box of the node. @zh 世界坐标系下包围盒。
```
Comment (SantyWang) [inline cocos/2d/framework/ui-transform.ts]: An axis aligned bounding box of this node in world space coordinate.

File: cocos/2d/renderer/mesh-buffer.ts [hunk 26]
```diff
@@ -34,75 +34,120 @@ interface IIARef {
...
+enum MeshBufferSharedBufferView {
     byteOffset,
     vertexOffset,
     indexOffset,
     dirty,
     count,
 }
 
+/**
+ * @en Mesh buffer used for 2d rendering, used internally and not of concern to the user
```
Comment (SantyWang) [inline cocos/2d/renderer/mesh-buffer.ts]: miss . and 。

File: cocos/2d/renderer/mesh-buffer.ts [hunk 27]
```diff
@@ -111,10 +156,15 @@ export class MeshBuffer {
     }
 
     protected _iData: Uint16Array = null!;
+    /**
+     * @en Indexes Data
```
Comment (SantyWang) [inline cocos/2d/renderer/mesh-buffer.ts]: The same like above

Comment (SantyWang) [reply cocos/2d/renderer/mesh-buffer.ts]: Indexes Data -> Indices data.

File: cocos/2d/renderer/stencil-manager.ts [hunk 28]
```diff
@@ -57,6 +66,11 @@ export enum StencilSharedBufferView {
     count,
 }
 
+/**
+ * @en Stencil state manager，user use may cause state misalignment
```
Comment (SantyWang) [inline cocos/2d/renderer/stencil-manager.ts]: ~~user use may cause state misalignment~~

File: cocos/2d/renderer/vertex-format.ts [hunk 29]
```diff
@@ -87,6 +97,18 @@ export const vfmtPosUvTwoColor = [
...
+ * @en Vertex format with the following layout
+ * 1. Vector 3 position attribute (Float32)
+ * 2. Vector 2 uv attribute (Float32)
+ * 3. First byte 4 color attribute (Byte)
+ * 4. Second byte 4 color attribute (Byte)
+ * @zh 包含以下数据的顶点格式
+ * 1. 三维位置属性（Float32）
+ * 2. 二维贴图 UV 属性（Float32）
+ * 3. 第一套 RGBA 颜色属性（Byte）
+ * 3. 第二套 RGBA 颜色属性（Byte）
```
Comment (SantyWang) [inline cocos/2d/renderer/vertex-format.ts]: 3 -> 4

File: cocos/2d/utils/dynamic-atlas/atlas-manager.ts [hunk 30]
```diff
@@ -273,6 +273,10 @@ export class DynamicAtlasManager extends System {
     }
 }
 
+/**
+ * @en Dynamic Atlas Manager，use to auto merge atlas
```
Comment (SantyWang) [inline cocos/2d/utils/dynamic-atlas/atlas-manager.ts]: merge atlas -> merge sprite frame

File: cocos/2d/assets/bitmap-font.ts [hunk 31]
```diff
@@ -102,33 +102,39 @@ export class FontAtlas {
  */
 @ccclass('cc.BitmapFont')
 export class BitmapFont extends Font {
+    /**
+     * @deprecated since v3.7.0, Useless Code.
```
Comment (SantyWang) [inline cocos/2d/assets/bitmap-font.ts:106]: Are you sure it's not working, or do you recommend not using it?

File: cocos/2d/assets/bitmap-font.ts [hunk 32]
```diff
@@ -102,33 +102,39 @@ export class FontAtlas {
...
-     * @en Font configuration
+     * @en Font configuration.
      * @zh 字体配置。
      */
     @serializable
     @editable
     public fntConfig: IConfig | null = null;
 
+    /**
+     * @deprecated since v3.7.0, Useless Code.
```
Comment (SantyWang) [inline cocos/2d/assets/bitmap-font.ts]: only for internal.

File: cocos/2d/assets/sprite-frame.ts [hunk 33]
```diff
@@ -475,8 +483,8 @@ export class SpriteFrame extends Asset {
     }
 
     /**
-     * @en Whether flip the uv in X direction
-     * @zh 延 X 轴方向, 翻转 UV
+     * @en Whether flip the uv in X direction.
+     * @zh 延 X 轴方向, 翻转 UV。
```
Comment (SantyWang) [inline cocos/2d/assets/sprite-frame.ts]: 延 -> 沿

File: cocos/2d/framework/ui-renderer.ts [hunk 34]
```diff
@@ -461,18 +477,27 @@ export class UIRenderer extends Renderer {
         }
     }
 
+    /**
+     * @deprecated Since v3.7
```
Comment (SantyWang) [inline cocos/2d/framework/ui-renderer.ts]: this is an engine private interface that will be removed in the future.

File: cocos/2d/framework/ui-renderer.ts [hunk 35]
```diff
@@ -461,18 +477,27 @@ export class UIRenderer extends Renderer {
...
+     */
     public setEntityColor (color: Color) {
         if (JSB) {
             this._renderEntity.color = color;
         }
     }
 
+    /**
+     * @deprecated Since v3.7
```
Comment (SantyWang) [inline cocos/2d/framework/ui-renderer.ts]: this is an engine private interface that will be removed in the future.

File: cocos/2d/assets/bitmap-font.ts (hunk 31 above)
Comment (LinYunMo) [reply cocos/2d/assets/bitmap-font.ts:106]: It's not working

**Review by SantyWang — APPROVED** [review]

**Pull request merged**